### DISCUSSION TITLE
How to changed the "underline--bar" default style "tall" to "solid"?

### DISCUSSION THREAD

**Discussion by Yurii-huang** [Textualize/textual#3716] [Q&A] [answered]
```
 Tabs:focus .underline--bar {
        background: $foreground 20%;
    }
```

![image](https://github.com/Textualize/textual/assets/34232854/0a50ba90-0c83-4928-b3cd-bf76a4906efb)

**Comment by TomJGooding** [comment]
I don't think it's possible to change the thickness/style of the underline bar, only the color. [The actual glyphs used to create the bar are baked into the 'Bar' renderable](https://github.com/Textualize/textual/blob/main/src/textual/renderables/bar.py#L38-L40) used by the Tabs widget by the looks of it.

**Reply by Yurii-huang** [reply]
This default style is not very user-friendly on serial ports or other high-resolution TTYs

![image](https://github.com/Textualize/textual/assets/34232854/a7b44580-db92-4836-8cda-ac8e28d90ad7)

**Reply by Yurii-huang** [reply]
Do you have a good suggestion for changing the style？

**Reply by TomJGooding** [reply, marked as answer]
Nothing straightforward I'm afraid. You'd need to create a custom [`Bar` renderable](https://github.com/Textualize/textual/blob/main/src/textual/renderables/bar.py), then a custom [`Underline` widget ](https://github.com/Textualize/textual/blob/main/src/textual/widgets/_tabs.py#L26)which renders your bar, then a custom [`Tabs` widget ](https://github.com/Textualize/textual/blob/main/src/textual/widgets/_tabs.py#L180) composed with your custom underline.

That's all assuming you aren't using `TabbedContent`, which would require extra work.
